qermit/spectral_filtering/spectral_filtering.py

```diff
@@ -0,0 +1,642 @@
+                [i for i in range(size)]
+                for size in sym_val_grid_list[0].shape
+            ]
+            for grid_point in product(*grid_point_val_list):
+                                                
+                # Generate dictionary mapping every symbol to it's value at
+                # the given point in the grid.
+                sym_map = {
+                    sym:sym_val_grid[grid_point]
+                    for sym_val_grid, sym in zip(sym_val_grid_list, sym_list)
+                }  
+                sym_dict = SymbolsDict().symbols_from_dict(sym_map)
+                circ = obs_exp.AnsatzCircuit.Circuit.copy()
+                anz_circ = AnsatzCircuit(
+                    circ,
+                    obs_exp.AnsatzCircuit.Shots,
+                    sym_dict
+                )
+                # This needs to be a deep copy
+                obs = deepcopy(obs_exp.ObservableTracker)
+                
+                obs_exp_grid[grid_point] = ObservableExperiment(anz_circ, obs)
+                                
+            obs_exp_grid_list.append(obs_exp_grid)
+        
+        return (obs_exp_grid_list, )
+    
+    return MitTask(
+        _label="ObsExpGridGen",
+        _n_out_wires=1,
+        _n_in_wires=2,
+        _method=task,
+    )
+
+
+def gen_symbol_val_gen_task(n_sym_vals:int) -> MitTask:
+    """Generates task which produces a grid of values taken by the symbols in
+    the circuit. The values are generated uniformly in the interval [0,2]
+    (factors of pi give full coverage of the Bloch sphere) for each symbol.
+    The points on the grid are equally spaces, and the number of values
+    is the same in each dimension.
+
+    :param n_sym_vals: The number of values to be taken by each symbol
+        in the circuit.
+    :type n_sym_vals: int
+    :return: Task generating grid of symbol values.
+    :rtype: MitTask
+    """
+    
+    def task(
+        obj,
+        obs_exp_list: List[ObservableExperiment]
+        ) -> Tuple[List[ObservableExperiment], List[List[NDArray]]]:
+        """Produces a grid of values taken by the symbols in
+        the circuit. The values are generated uniformly in the interval [0,2]
+        (factors of pi give full coverage) for each symbol. The points on the
+        grid are equally spaces, and the number of values is the same in
+        each dimension.
+
+        :param obs_exp_list: List of ObservableExperiments to be run.
+        :type obs_exp_list: List[ObservableExperiment]
+        :return: List of observable experiments is unchanged. For each
+            experiment, a grid is generated.
+        :rtype: Tuple[List[ObservableExperiment], List[List[np.ndarray]]]
+        """
+        
+        # A collection of symbol value lists is generated for each
+        # ObservableExperiment in obs_exp_list. Within each collection of
+        # lists, there is a list for each of the symbols.
+        sym_vals_list = []
```

Comment:
`
        return (
            obs_exp_list,
            [[np.linespace(0, 2, n_sym_vals, endpoint=False) for _  in range(len(obs_exp.AnsatzCircuit.SymbolsDict.symbols_list))] for obs_exp in obs_exp_list]
        ,)`

Reply:
sweet sweet list comprehension